cas/ctg_store.c

```diff
@@ -1028,27 +1057,57 @@ static int ctg_op_exec(struct m0_ctg_op *ctg_op, int next_phase)
 		m0_be_btree_insert_inplace(btree, tx, beop, key, anchor, zones);
 		break;
 	case CTG_OP_COMBINE(CO_GET, CT_BTREE):
+		if (use_ver) {
+			m0_be_btree_lookup_alive_inplace(btree, beop, key,
+							 anchor);
+			break;
+		}
+		/* XXX: sneaky fallthrough */
 	case CTG_OP_COMBINE(CO_GET, CT_META):
+		M0_ASSERT(!use_ver);
 		m0_be_btree_lookup_inplace(btree, beop, key, anchor);
 		break;
```

Comment:
Could you put a comment here saying that CT_BTREE-s will be versioned only.

Reply:
it took ~30seconds to understand the context.

Reply:
I added a bunch of pre-conditions instead.